.github/workflows/extract-args.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  workflow_call:
+    inputs:
+      file:
+        required: true
+        type: string
+      image:
+        required: true
+        type: string
+    outputs:
+      tags: 
+        description: "Image tags with all java versions appended to image input"
+        value: ${{ jobs.parse.outputs.tags }}
+      args: 
+        description: "Build args content read from input file"
+        value: ${{ jobs.parse.outputs.args }}
+
+jobs:
+  parse:
+    runs-on: ubuntu-22.04
+    outputs:
+      tags: ${{ steps.extract.outputs.tags }}
+      args: ${{ steps.extract.outputs.args }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        
+      - shell: bash
+        id: extract
+        run: |
+          TOTAL_JAVA_VERSION=$(cat ${{ inputs.file }} | grep "JAVA_VERSION" | cut -d'=' -f2)
+          BASE_JAVA_VERSION=$(echo $TOTAL_JAVA_VERSION | cut -d'-' -f1)
+          MAJOR_JAVA_VERSION=$(echo $BASE_JAVA_VERSION | cut -d'.' -f1)
+          echo "tags=${{ inputs.image }}:$MAJOR_JAVA_VERSION,${{ inputs.image }}:$BASE_JAVA_VERSION,${{ inputs.image }}:$TOTAL_JAVA_VERSION" >> $GITHUB_OUTPUT
+          {
+            echo "args<<nEOFn"
+            echo "$(cat ${{ inputs.file }})"
```

Comment:
wasted too much time here because of not wrapping this in echo not creating new line and leading to EOF not found 😔